pkg/operator/operator.go

```diff
@@ -147,13 +147,47 @@ func (optr *Operator) Run(workers int, stopCh <-chan struct{}) {
 		return
 	}
 	glog.Info("Synched up caches")
+	go wait.Poll(machineRolloutPollInterval, machineRolloutTimeout, func() (bool, error) {
+		return deployMachines(*optr)
+	})
+
 	for i := 0; i < workers; i++ {
 		go wait.Until(optr.worker, time.Second, stopCh)
 	}
 
 	<-stopCh
 }
 
+func deployMachines(optr Operator) (bool, error) {
+	operatorConfig, err := render.Config(optr.config)
+	if err != nil {
+		glog.Errorf("Error decoding operator config: %v", err)
+		panic(err)
```

Comment:
Why `panic`? Why not return `err`? Ditto for the other occurrences.

Reply:
These are two separate things. Sure, wait requires `bool, error`.

However reviewing this function I see that we always return `nil` for error - so as a reader/reviewe of the function I'm asking why does the signature allegedly return an error, when in reality it never (appears) to return one.

If this function is never going to return an error, then just change the caller to be:

```
	go wait.Poll(machineRolloutPollInterval, machineRolloutTimeout, func() (bool, error) {
		return deployMachines(*optr), nil
	})
```
because, in actuality, `deployMachines()` doesn't return error. We seem to have conflated the function signature to make it a one-line call from `wait.Poll()`.

Reply:
panic because there is no chance on retry err will get fixed. And not panic'ing at places where there is chance that condition will be met i.e deployment may appear in some time.